Lib/tempfile.py

```diff
@@ -900,18 +899,7 @@ def resetperms(path):
                         # raise NotADirectoryError and mask the PermissionError.
                         # So we must re-raise the current PermissionError if
                         # path is not a directory.
-                        try:
-                            st = _os.lstat(path)
-                        except OSError:
-                            if ignore_errors:
-                                return
-                            raise
-                        if (_stat.S_ISLNK(st.st_mode) or
-                            not _stat.S_ISDIR(st.st_mode) or
-                            (hasattr(st, 'st_file_attributes') and
-                             st.st_file_attributes & _stat.FILE_ATTRIBUTE_REPARSE_POINT and
-                             st.st_reparse_tag == _stat.IO_REPARSE_TAG_MOUNT_POINT)
-                        ):
+                        if not _os.path.isdir(path) or _os.path.isjunction(path):
```

Comment:
I assume "TOCTOU" issues are out-of-scope here (both in the original implementation and the simplified one) since it is perfectly fine to handle any concurrent modification of the current directory with "undefined behaviour")?

Reply:
It is the best that we can do with the current API. I will consider supporting a new error handler interface that accepts `dir_fd` and the relative path.